nservicebus/sagas/concurrency.md

```diff
@@ -1,48 +1,58 @@
 
-Concurrent access to saga instances is divided into two scenarios;
+NOTE: When using some transports, the [outbox](/nservicebus/outbox/) must be used in order to provide this level of consistency.
 
- * Concurrently trying to create the same instance of a new saga.
- * Concurrently trying to update the same instance of an existing saga.
+Saga state is created when a saga is started by a message. It may be updated when the saga handles another message. It is deleted when the saga handles another message which completes the saga. In any of these cases, multiple messages may be handled simultaneously.
 
-Each saga persister will honor the following semantics, see the specific persister documentation for implementation details.
+## Starting a saga
 
-## Concurrent access to non-existing saga instances
+When messages that start the same logical saga are handled simultaneously, NServiceBus ensures that only one message starts the saga. This is done in the same transaction as receiving the message and sending new messages.
 
-Sagas are started by the message types that a handled with `IAmStartedByMessages<T>`. If messages mapped to the same saga instance are processed concurrently there is a risk that duplicates of the instance will be created.
-
-In this case only one message is allowed to complete processing. The others roll back and the built-in retries in NServiceBus kick in. On the next retry, the saga instance is found, the race condition is solved, and that saga instance is updated instead, see below.
+Only one message handler is allowed to succeed. The saga state is created and sent messages are dispatched. The other message handlers fail, roll back, and their messages enter [recoverability](/nservicebus/recoverability/). When those messages are retried, the existing saga state is found. Handling the messages may involve changing the state or completing the saga. See below for how those scenarios are handled.
 
 partial: unique
 
-## Concurrent access to existing saga instances
+## Changes to saga state
+
+When messages are handled simultaneously by the same logical saga, and involve changing the saga state, NServiceBus ensures that only one message handler changes the state in the same transaction as receiving the message and sending any new messages.
```

Comment:
The "same transaction" language that's used here and other places in the document is sticking out a bit to me. What exactly is trying to be conveyed here? Why bring transactions into it at all?

Reply:
Agreed. "transaction" is a bit overloaded, and could confuse things. I'll find an alternate term/phrase.